src/utils.ts

```diff
@@ -3,53 +3,76 @@ import {
   DosageInstruction,
   Medication,
   MedicationDispense,
-  MedicationDispenseStatus,
   MedicationReferenceOrCodeableConcept,
   MedicationRequest,
-  MedicationRequestMedicationDispenseCombinedStatus,
+  MedicationRequestCombinedStatus,
+  MedicationRequestFulfillerStatus,
   MedicationRequestStatus,
   Quantity,
 } from "./types";
 import { fhirBaseUrl, parseDate } from "@openmrs/esm-framework";
 import {
-  OPENMRS_FHIR_EXT_MEDICINE,
   OPENMRS_FHIR_EXT_DISPENSE_RECORDED,
+  OPENMRS_FHIR_EXT_MEDICINE,
+  OPENMRS_FHIR_EXT_REQUEST_FULFILLER_STATUS,
 } from "./constants";
 import dayjs from "dayjs";
 
 /**
- * This is a bit of a mouthful, but, within the UI, the "status" of a request we want to display to the pharmacist is
- * a combination of the status of the request and the status of the most recent dispensing event; given a request
- * status and a dispense status, this calculates the actual status we want to display to the pharmacist
+ * Within the UI, the "status" of a request we want to display to the pharmacist is
+ * a combination of the status and the fulfiller statuts; given a request
+ * this calculates the actual status we want to display to the pharmacist
```

Comment:
here's the updated calculation for the "combined status" which I talked about in my comment above.